aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/TestConfiguration.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.AmplifyConfiguration;
+import com.amplifyframework.core.InitializationStatus;
+import com.amplifyframework.core.category.CategoryType;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.testutils.Await;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.client.Callback;
+import com.amazonaws.mobile.client.UserStateDetails;
+import org.json.JSONException;
+
+final class TestConfiguration {
+    private static TestConfiguration singleton;
+    private final AWSS3StoragePlugin plugin;
+    private final String bucket;
+
+    private TestConfiguration(Context context) throws AmplifyException {
+        plugin = new AWSS3StoragePlugin();
+        bucket = getBucketNameFromPlugin(context, plugin);
+
+        Amplify.addPlugin(plugin);
+        configureAmplify(context);
+        setUpCredentials(context);
+    }
+
+    /**
+     * Process-wide configuration for the Storage instrumentation tests.
+     * @return A TestConfiguration instance
+     */
+    @NonNull
+    static synchronized TestConfiguration configureIfNotConfigured() throws AmplifyException {
+        if (singleton == null) {
+            singleton = new TestConfiguration(ApplicationProvider.getApplicationContext());
+        }
+        return singleton;
+    }
+
+    private static void configureAmplify(Context context) {
+        // Configure Amplify, and wait for Storage to be ready.
+        Await.result((onResult, onError) -> {
+            // Listen for initialization success messages
+            Amplify.Hub.subscribe(HubChannel.STORAGE, event -> {
+                // When we get one, end the Await call by firing a result
+                if (InitializationStatus.SUCCEEDED.toString().equals(event.getName())) {
```

Comment:
I know I have this stuff in the DataStore. But, we really shouldn't need it. We should have the source code itself enqueue operations to execute _after_ receiving this event. That way, the caller doesn't need to wait for this event. The caller can just start enqueing operations.

This is a much larger architectural problem, that is not really related to your work. I'm just worried about this pattern spreading into more tests (even though I started it.)

At minimum, we should leave a note at the top of all this like, "TODO: wait for initialization in source code, not in test. Otherwise, a customer would have to use this await block, too...."

Reply:
Ah okay, I'll replace it with simple `Amplify.configure()`. I thought it was necessary to include this.